File: frontend/src/types/orientation.ts
export type OrientationOption = {
  degrees: number
  orientation: 'portrait' | 'landscape'
  preview_url: string
}

export type OrientationDataResponse = {
  private_id: string
  score_orientation: 'portrait' | 'landscape'
  current_rotation_degrees: number
  current_orientation: 'portrait' | 'landscape'
  rotation_options: OrientationOption[]
  reimport_in_progress: boolean
  reimport_progress: number
  reimport_error: string | null
  reimport_error_message: string | null
}

File: api/app/schemas/orientation.py
from typing import Literal

from pydantic import BaseModel, Field


class OrientationOption(BaseModel):
    degrees: int
    orientation: Literal["portrait", "landscape"]
    preview_url: str


class OrientationDataResponse(BaseModel):
    private_id: str
    score_orientation: Literal["portrait", "landscape"]
    current_rotation_degrees: int = 0
    current_orientation: Literal["portrait", "landscape"]
    rotation_options: list[OrientationOption]
    reimport_in_progress: bool = False
    reimport_progress: float = 0.0
    reimport_error: str | None = None
    reimport_error_message: str | None = None


class ReorientRequest(BaseModel):
    rotation_degrees: int = Field(ge=0, le=270)


class ReorientResponse(BaseModel):
    status: str
    job_id: str


class RetryPageCacheResponse(BaseModel):
    status: str = "started"
